scripts/files/fs.py

```diff
@@ -42,3 +48,64 @@ def exists(path: str) -> bool:
     if is_s3(path):
         return fs_s3.exists(path)
     return fs_local.exists(path)
+
+
+def _read_write_file(target: str, file: str) -> str:
+    """write a file to a specificed target dir.
+
+    Args:
+        target: target directory path
+        file: file to read-write
+
+    Returns:
+        written file path
+    """
+    download_path = os.path.join(target, f"{file.split('/')[-1]}")
+    get_log().info("Read-Write File", path=file, target_path=download_path)
+    write(download_path, read(file))
+    return download_path
+
+
+def write_all(inputs: List[str], target: str, concurrency: Optional[int] = 10) -> List[str]:
+    """Writes list of files to target destination using multithreading.
+
+    Args:
+        inputs: list of files to read
+        target: target folder to write to
+
+    Returns:
+        list of written file paths
+    """
+    written_tiffs: List[str] = []
+    with ThreadPoolExecutor(max_workers=concurrency) as executor:
+        futuress = {executor.submit(_read_write_file, target, input): input for input in inputs}
+        for future in as_completed(futuress):
+            if future.exception():
+                get_log().warn("Failed Read-Write", error=future.exception())
+            else:
+                written_tiffs.append(future.result())
+
+    if len(inputs) != len(written_tiffs):
+        get_log().error("Missing Files", missing_file_count=len(inputs) - len(written_tiffs))
```

Comment:
You can simplify `missing_file_count` to `count` as we know the context with the error message